src/main/scala/io/pinecone/spark/pinecone/PineconeOptions.scala

```diff
@@ -12,16 +12,23 @@ class PineconeOptions(config: CaseInsensitiveStringMap) extends Serializable {
 
   val apiKey: String = getKey(PINECONE_API_KEY_CONF, config)
   val indexName: String = getKey(PINECONE_INDEX_NAME_CONF, config)
+  val sourceTag: String = getSourceTag(config)
 
   private def getKey(key: String, config: CaseInsensitiveStringMap): String = {
     Option(config.get(key)).getOrElse(
       throw new RuntimeException(s"Missing required parameter $key")
     )
   }
+
+  private def getSourceTag(config: CaseInsensitiveStringMap): String = {
+    val value = Option(config.get(PINECONE_SOURCE_TAG_CONF)).getOrElse("")
+    s"spark_$value"
+  }
```

Comment:
It seems like "getKey" is intended to retrieve required params, while "getSourceTag" is an optional one. 

Consider making them both generic as something like `getRequiredConfig(name, config)` and `getOptionalConfig(name, config, default)`?

Reply:
I initially thought about it but it didn't make sense for just one variable, so maybe in future